hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v2/ActiveTimelineV2.java

```diff
@@ -524,12 +556,14 @@ protected <T> void transitionStateToComplete(boolean shouldLock, HoodieInstant f
           throw new HoodieIOException(
               "Could not rename " + fromInstantPath + " to " + toInstantPath);
         }
+        return instantWithCompletionTime;
       } else {
         // Ensures old state exists in timeline
         ValidationUtils.checkArgument(
             metaClient.getStorage().exists(getInstantFileNamePath(fromInstantFileName)),
             "File " + getInstantFileNamePath(fromInstantFileName) + " does not exist!");
-        createCompleteFileInMetaPath(shouldLock, toInstant, metadata);
+        String completionTime = HoodieInstantTimeGenerator.formatDateBasedOnTimeZone(new Date(createCompleteFileInMetaPath(shouldLock, toInstant, metadata)));
```

Comment:
nts: need to ensure this is consistent with our approach to time generation

Reply:
I think this is fine. keeping this open.. and tracking. 

Will get @the-other-tim-brown 's eyes on this next week.